functors_task_1/student/include/impl.h

```diff
@@ -16,16 +25,23 @@ struct BeerOrganizer
  *
  * @note Only Corona and HoeGaarden are expensive
  */
-bool isExpensiveBeer(/**???*/)
+bool isExpensiveBeer(const BeerBrand beerBrand)
 {
+    return (BeerBrand::Corona==beerBrand || BeerBrand::HoeGaarden==beerBrand);
 }
 
 /**
  * @todo Implement lambda beer country equality comparator
  * @return true if beer county is the same, false otherwise
  */
-auto sameCountry = [](/**???*/)
+auto sameCountry = [](const BeerBrand lhs,const BeerBrand rhs)
 {
+    const auto lCountry= getBeerCountry(lhs);
+    const auto rCountry= getBeerCountry(rhs);
+    if(Country::Unknown==lCountry){
```

Comment:
Why do you think that false should be returned?
Also, why rhs county is not checked?

Reply:
descripion: return true if beer county is the same, false otherwise

Country::Unknown  - it's not a country, it union of some others country
that is why one country of one Country::Unknown != other Country::Unknown

checkin rhs is non nessessary becouse if lhs not Unknown and rhs Unknown false will be returned

Reply:
"Unknown" represents a special invalid entry. This comparator should not care about invalid entries at all. It should just check for equality of properties, even if they are both invalid.

Reply:
there is many invalid variant can be
so there is has to be checking if country invalid
and if country are the same.

Reply:
it shouldn't be done by comparator